src/category_theory/elements.lean

```diff
@@ -1,17 +1,31 @@
-import category_theory.types
-import category_theory.comma
 import category_theory.equivalence
+import category_theory.comma
 import category_theory.punit
 import category_theory.eq_to_hom
 
+/-!
+Copyright (c) 2019 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+
+The category of elements, also known as the Grothendieck construction.
+
+Given a functor `F : C ⥤ Type`, an object of `F.elements` is a pair `(X : C, x : F.obj X)`.
+A morphism `(X, x) ⟶ (Y, y)` is a morphism `f : X ⟶ Y` in `C`, so `F.map f` takes `x` to `y`.
+-/
+
 namespace category_theory
 
 universes v u
 variables {C : Type u} [𝒞 : category.{v} C]
 include 𝒞
 
+/-- The type of objects for the category of elements of a functor `F` is a pair `(X : C, x : F.obj X)`. -/
```

Comment:
This seems redundant with the definition itself.

Reply:
@robertylewis, what's the intention of the doc guidelines here? Should I remove this comment?

Reply:
I think this is fine. It clarifies that this is defining the type of objects for a particular category. I don't mind the redundancy. It might be nice to reference `C` in the doc string, since the variable won't be visible when this is seen as a tooltip.